src/Pos/ExecMode.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+module Pos.ExecMode
+    ( ExecMode(..)
+    , ExecModeM
+    ) where
+
+import           Universum
+
+import           Control.Monad.Base          (MonadBase)
+import           Control.Monad.Fix           (MonadFix)
+import qualified Control.Monad.Reader        as Mtl
+import           Control.Monad.Trans.Control (MonadBaseControl (..))
+import           Data.Coerce                 (coerce)
+import qualified Ether
+import           Ether.Internal              (HasLens (..))
+import           Mockable                    (ChannelT, Counter, Distribution, Gauge,
+                                              MFunctor' (..), Mockable (..), Promise,
+                                              SharedAtomicT, SharedExclusiveT, ThreadId)
+import           System.Wlog                 (CanLog)
+
+import           Pos.Util.Util               (PowerLift (..))
+
+type family ExecModeM mode :: * -> *
+
+newtype ExecMode mode a = ExecMode { unExecMode :: ExecModeM mode a }
```

Comment:
Furthermore, it has a special `Ether.MonadReader` instance that doesn't do propagation. It's safe because `ExecMode` is always the base layer, and this allowed me to delete the `Tags` instances for `NodeContext` (which were rather atrocious).